doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,161 @@
+``load_sample`` function; this works with ``load`` as well.
+
+.. code-block:: python
+
+   ds = yt.load_sample("Gadget3-snap-format2", index_order=(5, 5))
+
+So this is how you *change* the index order, but it doesn't explain precisely
+what this "index order" actually is.
+
+Index Order
+-----------
+
+The bitmap index system is based on a two-level scheme for assigning positions
+in three-dimensional space to integer values.  What this means is that each
+particle is assigned a "coarse" index, which is global to the full domain of
+the collection of particles, and *if necessary* an additional "refined" index
+is assigned to the particle, within that coarse index.
+
+The index "order" values refer to the number of entries on a side that each
+index system is allowed.  For instance, if we allow the particles to be
+subdivided into 8 "bins" in each direction, this would correspond to an index
+order of 3 (as $2^3 = 8$); correspondingly, an index order of 5 would be 32
+bins in each direction, and an index order of 7 would be 128 bins in each
+direction.  Each particle is then assigned a set of i, j, k values for the bin
+value in each dimension, and these i, j, k values are combined into a single
+(64-bit) integer according to a space-filling curve.
+
+The process by which this is done by yt is as follows:
+
+  1. For each "chunk" of data -- which may be a file, or a subset of a file in
+     which particles are contained -- assign each particle to an integer value
+     according to the space-filling curve and the coarse index order.  Set the
+     "bit" in an array of boolean values that each of these integers correspond
+     to.  Note that this is almost certainly *reductive* -- there will be fewer
+     bits set than there are particles, which is *by design*.
+  2. Once all chunks or files have been assigned an array of bits that
+     correspond to the places where, according to the coarse indexing scheme,
+     they have particles, identify all those "bits" that have been set by more
+     than one chunk.  All of these bits correspond to locations where more than
+     one file contains particles -- so if you want to select something from
+     this region, you'd need to read more than one file.
+  3. For each "collision" location, apply a *second-order* index, to identify
+     which sub-regions are touched by more than one file.
+
+At the end of this process, each file will be associated with a single "coarse"
+index (which covers the entire domain of the data), as well as a set of
+"collision" locations, and in each "collision" location a set of bitarrays that
+correspond to that subregion.
+
+When reading data, yt will identify which "coarse" index regions are necessary
+to read.  If any of those coarse index regions are covered by more than one
+file, it will examine the "fine" index for those regions and see if it is able
+to subset more efficiently.  Because all of these operations can be done with
```

Comment:
It seems to me that the coarse index operates like a root grid and the fine index operates sort of like AMR of 1-level deep.  But I think this means that it is possible to overload an index with too many particles in some circumstances, depending on the indexing orders chosen, right?  Is this ever a concern or would it choose a second fine index for a second level of AMR?  Or would it just error out?

If I choose, (3,2) as my index order for a 3D dataset, I think it means that my coarse array supports 2**3=8 bins on any side, so 8**3 = 512 total index locations.  And then my refined index supports 2**2=4 bins on any side, so 4**3 = 64 index locations within that.  So it would only support 512*64 = 32k particles?  Is that right, or am I totally off base?  

Based on the values you have chosen as default (7,5), it seems like this won't be a problem: 2**7**3 * 2**5**3 = 2**36 ~ 7e10, but with (5,3): 2**5**3 * 2**3**3 = 2**24 ~ 2e7 it might be.

Reply:
This will not happen.  The indexing is not used for anything other than choosing which files/chunks to read particles from, and then these particles are validated using the standard selection methods.

The only danger of having too coarse of an index is that too many particles will be read, found to be outside the region of interest, and discarded.  The particles are still all regarded as discrete points (or smoothed points) individually and validated that way, and the indexing system only attempts to reduce the amount of overzealous IO that precedes that validation.

You could in principle index with (0, 0).  It would just accomplish nothing, because every chunk would match for every selection, and so yt would read every single particle in the simulation, validate it's within the selector, and then either retain or discard it.

Reply:
OK, I guess I was thinking of the coarse/refined indexing as an octree/kd-tree like structure, as to where the particle locations were stored and accessed.  But that seems that's not quite right here, because I'd think that when you get to the "leaf" of the tree, it's just a single particle.  I guess I'm thinking of the wrong paradigm.

Reply:
I've added a bit to the text about why we're doing the indexing to begin with, which I hope will clear this up.  Let me know what you think; it's in the most recent commit.